framework/src/base/Registry.C

```diff
@@ -110,17 +110,23 @@ Registry::addKnownLabel(const std::string & label)
 }
 
 void
-Registry::addDataFilePath(const std::string & fullpath)
+Registry::addDataFilePathFromApp(const std::string & fullpath)
```

Comment:
This is unrelated to the linked issue and I strongly recommend against it as it messes up the design,m which is explicitly based on an automatically (if present) installed `data` dir.

Reply:
I disagree. You called this function anyway. Maybe you or me misunderstood something?

Reply:
This function may not work with binary installation. Maybe I missed something because I cannot find codes for auto installed data dir.

Reply:
No, you don't need to call `addDataFilePath` directly. The design does not require it. What you need can be done without these changes.